jetty-server/src/main/java/org/eclipse/jetty/server/HttpConnection.java

```diff
@@ -302,15 +301,14 @@ else if (filled < 0)
         {
             if (LOG.isDebugEnabled())
                 LOG.debug("{} caught exception {}", this, _channel.getState(), x);
-            BufferUtil.clear(_requestBuffer);
```

Comment:
I don't think this `clear()` can be removed.
If you don't clear the buffer, then it's leaked because `releaseRequestBuffer()` does not release it if it still has content.

Reply:
You're right, but I replaced the call to `releaseRequestBuffer` with an explicit null check and release to make it more clear that this error handling is a bit of a special case.